### ISSUE TITLE
[SORMAS-SB] Contacts > Contact Directory > Follow-up Visits > Filter by "Visit Remarks"

### ISSUE THREAD

**Issue by Candice-Louw** [hzi-braunschweig/SORMAS-Project#4506]
### Situation Description
As a contact tracer, I would like to be able to read the comments that contact persons report via the symptom diary.

![image](https://user-images.githubusercontent.com/65529128/108741491-16235f00-7537-11eb-906d-e8b0ac1ebddc.png)

In order to do this, I need to be able to filter for "Visit Remarks" that are filled out (i.e. if "Visit Remarks" field is not empty [!=""] for a certain contact person in the follow-up view, show them in the directory).

### Feature Description
Introduce an additional filter "Visit Remarks", as a checkbox on the Contacts Directory in the Follow-up Visits view.
![image](https://user-images.githubusercontent.com/65529128/108741699-479c2a80-7537-11eb-9a3a-3c93c36a6bc7.png)

### Possible Alternatives
None known of - please feel free to suggest.

### Additional Information
This is for Contacts now, although SORMAS-SB will be expanded to Cases soon too and this should please be kept in mind.

**Comment by bernardsilenou** [comment]
@Candice-Louw  
- This is a complicated one since the remark is a free txt variable
- Maybe we add a free txt search + filter, to look  for all variables having the suggested word: eg if someone type 'late' in the filter, then it would search the appearance of the word  in : visit remark, comments (some users can write their visit remarks in comments)
- We can also extend this to the case directory to search all free txt varaible (followup status comment, general comment, ...)  related to cases, etx

**Comment by Candice-Louw** [comment]
@bernardsilenou - the request here is not for searching the actual values in the text boxes (any comment box), but rather to see which users have submitted a comment (i.e. there is some content in the box).

For example-
As a contact tracer I am asked to perform the following:
Filter for all contacts, on the contacts directory (follow-up visits view), who have submitted a response through their symptom diary app to SORMAS, and who have left a comment for me to read/respond to, in the date range of 'today'.

I would filter for these values, read the comment of the KPs, and perform a certain action, based on their comment.  But at the moment, it is not possible to see if a comment was submitted with the questionnaires and contact tracers are not able to see whether there are questions directed at them.

The content of the comment in this context doesn't matter - the fact of the matter is that I want to know that a comment has been submitted (i.e. comment field is not empty ( !="" or <>"" )).

The search in all free texts proposal is great, but a separate issue from this one.

**Comment by bernardsilenou** [comment]
@Candice-Louw  I got it now !
- This makes it easier, then the `check box` option looks cool.

**Comment by Candice-Louw** [comment]
Cool, thank you for the feedback! 👍  Agreed on the checkbox. ✅

**Comment by MateStrysewske** [comment]
@Candice-Louw @bernardsilenou If I'm not mistaken and don't misunderstand you, what you need here can be achieved by simply sorting the list by visit remarks; one of the sorting orders (not sure whether it's ascended or descended) will always list those with empty remarks first. Would that be sufficient?

**Comment by Candice-Louw** [comment]
For the individual Case or Contact's Follow-up Visits, this works well, yes.

The request is to have this on the Case and Contacts Directory, in Follow-up view, where the series of follow-ups is listed for multiple cases/contacts, without knowing which one has in fact got a comment attached to it.  There is no "Visit Remarks" column to order by in this view (comments are associated with an individual visit, rather than the collection of visits as displayed in the directory).

Adding the checkbox would still be required, please.

**Comment by soja-hzi** [comment]
@Candice-Louw why is this SB specific and not a general feature for SORMAS?

**Comment by MateStrysewske** [comment]
@soja-hzi I don't think this issue will be specific to SORMAS-SB but that's just where the request is coming from

**Comment by bernardsilenou** [comment]
@kwa20 @Candice-Louw  Can we take this issue for internal refinement.
- The reason is that we have realized that when it comes to management of followup of contacts and even cases, users think in terms of persons, thus this implementation may not solve their need in case a person belongs to 2 contacts that do not overlap 100% in terms of followup time.
- The possible refinement needed my be to make this filter works in terms of person ie something like all to search for  all  contacts of the person with same disease having a comment.
- Even though we document data in terms of entities, we are thinking of implementing most of the filters and event statistics in terms of person as default. + option to compute in terms of entities.

**Comment by FredrikSchaeferVitagroup** [comment]
Seems to be covered with #4515, or?

**Comment by Candice-Louw** [comment]
@FredrikSchaeferVitagroup - yes, thank you. Please feel free to close this issue.
